src/parser/types/statement_node.rs

```diff
@@ -30,6 +30,8 @@ pub enum Statement {
     /// DIM A, B, C
     Dim(DimList),
 
+    Redim(DimList),
```

Comment:
Add tests:

- Cannot REDIM already DIMed variables
- Cannot change type via REDIM
- Can change array dimensions
- Cannot change number of dimensions
- Data is not preserved